// @ekoopensource/sdk/build/eos-sdk-js/src/utils/requests.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const axios_1 = require("axios");
const error_1 = require("./error");
const request = (method, path, options) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        headers: options.headers,
        baseURL: options.url,
        timeout: options.timeout,
        method: method,
        url: path,
        data: options.data
    };
    return axios_1.default(requestOptions)
        .then(response => response.data)
        .catch(err => errorHanding(err));
});
function errorHanding(err) {
    if (err.response !== null && err.response !== undefined && err.response.data !== null) {
        return err.response.data;
    }
    const errorMessages = [];
    errorMessages.push(error_1.default(err));
    return { errors: errorMessages };
}
exports.default = request;
